src/main/java/software/amazon/cloudformation/resource/ResourceTypeSchema.java

```diff
@@ -153,6 +149,57 @@ public ResourceTypeSchema(Schema schema) {
             return null;
         });
+
+        this.unprocessedProperties.computeIfPresent("tagging", (k, v) -> {
+            ((Map<?, ?>) v).forEach((key, value) -> {
+                if (!key.equals("tagProperty") && !key.equals("tagPropertyType")) {
+                    this.tagging.put(key.toString(), Boolean.valueOf(value.toString()));
+                } else if (key.equals("tagProperty")) {
+                    this.tagging.put(key.toString(), new JSONPointer(value.toString()));
+                } else {
+                    this.tagging.put(key.toString(), value.toString());
+                }
+            });
+
+            if (this.tagging.containsKey("taggable") && this.tagging.containsKey("tagOnCreate")
+                && this.tagging.get("taggable").equals(false) && this.tagging.get("tagOnCreate").equals(true)) {
+                throw new ValidationException("Invalid tagOnCreate value since taggable is marked false", "tagging",
+                                              "#/tagging/tagOnCreate");
+            } else if (this.tagging.containsKey("taggable") && this.tagging.containsKey("tagUpdatable") &&
+                this.tagging.get("taggable").equals(false) && this.tagging.get("tagUpdatable").equals(true)) {
+                throw new ValidationException("Invalid tagUpdatable value since taggable is marked false", "tagging",
+                                              "#/tagging/tagUpdatable");
+            } else if (this.tagging.containsKey("taggable") && this.tagging.containsKey("CFNSystemTags") &&
+                this.tagging.get("taggable").equals(false) && this.tagging.get("CFNSystemTags").equals(true)) {
+                throw new ValidationException("Invalid CFNSystemTags value since taggable is marked false", "tagging",
+                                              "#/tagging/CFNSystemTags");
+            } else if (this.tagging.containsKey("tagUpdatable") && this.tagging.get("tagUpdatable").equals(true)
+                && !this.handlers.containsKey("update")) {
+                throw new ValidationException("Invalid tagUpdatable value since update handler is missing", "tagging",
+                                              "#/tagging/tagUpdatable");
+            }
+            return null;
+        });
+
+        if (this.unprocessedProperties.containsKey("taggable")) {
+            this.taggable = Boolean.parseBoolean(this.unprocessedProperties.get("taggable").toString());
+            if (!this.tagging.containsKey("taggable")) {
+                this.tagging.put("taggable", this.taggable);
+                if (this.taggable) {
+                    // default value to true when taggable to enforce tagging feature
+                    tagging.put("tagOnCreate", true);
+                    tagging.put("tagUpdatable", true);
+                    tagging.put("CFNSystemTags", true);
+                }
+            } else {
+                if (!this.tagging.get("taggable").equals(this.taggable)) {
```

Comment:
I would suggest we should have stricter rule. Only one `taggable` should be allowed to use. Even they are same value, we should enforce them to configure one.  In this way, we are deprecating the previous `taggable` config.

Reply:
that make sense, will add the validation